### DISCUSSION TITLE
Нужен ли SplitLayout для одноколоночного приложения?

### DISCUSSION THREAD

**Discussion by mahnunchik** [VKCOM/VKUI#3566] [Q&A] [answered]
Необходимо ли использовать `SplitLayout` в одноколоночном приложении размещённого на платформе vk mini apps?

Исходя из текущей документации https://vkcom.github.io/VKUI/#!/Adaptivity приложение должно иметь следующую структуру:

```jsx
<ConfigProvider>
    <AdaptivityProvider>
        <AppRoot>
            <SplitLayout>
                <SplitCol>
                    <View>
                        <Panel >
                        </Panel>
                    </View>
                </SplitCol>
            </SplitLayout>
        </AppRoot>
    </AdaptivityProvider>
</ConfigProvider>
```

Но кажется более простым использование структуры:

```jsx
<ConfigProvider>
    <AdaptivityProvider>
        <AppRoot>
            <Root>
                <View>
                    <Panel >
                    </Panel>
                </View>
                <View>
                    <Panel>
                    </Panel>
                </View>
            </Root>
        </AppRoot>
    </AdaptivityProvider>
</ConfigProvider>
```

Собственно вопрос: какая структура более подходящая, что теряется при не использовании `SplitLayout`.

**Comment by mahnunchik** [comment]
Похожий тикет, который закрыли оставив без ответа https://github.com/VKCOM/VKUI/issues/2020

**Comment by mahnunchik** [comment]
Из документации не понятна связть между `SplitLayout`, `SplitCol` и `Root`.

* Нужно ли помещать `SplitLayout` в компонент `Root`?
* Или же наоборот в `SplitCol` нужно вложить `Root` со своим набором View/Panel?

**Reply by inomdzhon** [reply]
> - Нужно ли помещать SplitLayout в компонент Root?

Сделать

```
<Root>
  <SplitLayout>...</SplitLayout>
</Root>
```

в целом невозможно, потому что содержимым `Root` может быть только `View`.

> - Или же наоборот в SplitCol нужно вложить Root со своим набором View/Panel?

Да, именно так 👍

**Comment by inomdzhon** [comment, marked as answer]
Привет!

> Необходимо ли использовать SplitLayout в одноколоночном приложении размещённого на платформе vk mini apps?
> ...
> Собственно вопрос: какая структура более подходящая, что теряется при не использовании SplitLayout.

Если в приложении есть всплывающие окна, то без `SplitLayout` никак, т.к. у именно у него есть св-ва `modal` и `popout`.

PS1: Эти св-ва повторяются у `View`, но они `depracated`, с версии `v5` будет только у `SplitLayout`.
PS2: В целом систему лэйаута в библиотеке надо переработать, после `v5` будем думать над этим.

**Comment by mahnunchik** [comment]
@inomdzhon Спасибо за разъяснения!

Я правильно понимаю, что полная структура приложения в текущей архитектуре это:

```js
<ConfigProvider>
    <AdaptivityProvider>
        <AppRoot>
            <SplitLayout modal="" popout=""> // тут у нас все всплывающие окна
                <SplitCol>
                    <Root> // Root может не быть, если нужно только 1 View
                        <View>
                             <Panel>
                             </Panel>
                        </View>
                    </Root>
                </SplitCol>
            </SplitLayout>
        </AppRoot>
    </AdaptivityProvider>
</ConfigProvider>
```

Очень просятся разъяснения в документацию на следующих страницах:

* https://vkcom.github.io/VKUI/#/SplitLayout
* https://vkcom.github.io/VKUI/#/Root
* https://vkcom.github.io/VKUI/#/AppRoot

А то сейчас по документации, как будто каждый из них может выступать root-ом приложения.

**Reply by inomdzhon** [reply]
Структура верная 👍 

По поводу разъяснений в документации, подумаем как поправить. Отметил себе, потом создам issue. Спасибо за замечания 🙏 ❤️

**Comment by mahnunchik** [comment]
И ещё вопрос, я правильно понимаю, что для одноколоночного приложения, которое будет использовано только на платформе vkminiapps подходит следующий лайаут без модофикаторов на ширину/колонки и прочее:

```js
<SplitLayout>
    <SplitCol>
    // тут всё приложение
    </SplitCol>
</SplitLayout>

**Reply by inomdzhon** [reply]
Да, всё так 👍
